Penteract/Assets/Scripts/LasersGenerator.cpp

```diff
@@ -0,0 +1,95 @@
+#include "LasersGenerator.h"
+
+#include "GameplaySystems.h"
+#include "GameObject.h"
+
+EXPOSE_MEMBERS(LasersGenerator) {
+    MEMBER(MemberType::FLOAT, coolDownOn),
+    MEMBER(MemberType::FLOAT, coolDownOnTimer),
+    MEMBER(MemberType::FLOAT, coolDownOff),
+    MEMBER(MemberType::FLOAT, coolDownOffTimer),
+    MEMBER(MemberType::GAME_OBJECT_UID, laserTargetUID),
+    MEMBER(MemberType::GAME_OBJECT_UID, pairGeneratorUID)
+};
+
+GENERATE_BODY_IMPL(LasersGenerator);
+
+void LasersGenerator::Start() {
+    GameObject* owner = &GetOwner();
+    GameObject* pair = GameplaySystems::GetGameObject(pairGeneratorUID);
+
+    if (owner) {
+        animationComp = owner->GetComponent<ComponentAnimation>();
+    }
+    if (pair) {
+        pairAnimationComp = pair->GetComponent<ComponentAnimation>();
+        pairScript = GET_SCRIPT(pair, LasersGenerator);
+    }
+
+    laserObject = GameplaySystems::GetGameObject(laserTargetUID);
+    if (laserObject) {
+        laserObject->Disable();
+    }
+}
+
+void LasersGenerator::Update() {
+    if (!animationComp) {
+        Debug::Log("animationComp is nullptr");
```

Comment:
Though this is a really good idea, and we should really use these kind of logs when we're missing references, it hasn't been done in any other script and this would probably spam the console if the Animation Component is missing. So perhaps, this should just be an `if(!animationCop) return;` because I'm guessing laser behavior should be cancelled if there's no animator! Idk, you tell me what's expected! (**It would also throw a null pointer exception when accessing it in the switch below**).

Reply:
Yes, you are completely right. I'm changing it for a return.